### ISSUE TITLE
MonadIO instance

### ISSUE THREAD

**Issue by chris-martin** [josefs/STMonadTrans#29]
Would a pull request adding a `MonadIO` instance be welcome?

```haskell
instance (MonadIO m) => MonadIO (STT s m) where
    liftIO x = lift (liftIO x)
```

This would require raising the minimum `base` version to `4.9` (GHC 8.0, released in 2016).

**Comment by andreasabel** [comment]
> This would require raising the minimum base version to 4.9 (GHC 8.0, released in 2016).

Or using a conditional...

**Comment by chris-martin** [comment]
What's this package's policy for when to drop support for old versions?

**Comment by andreasabel** [comment]
We could also explicitly depend on `transformers`, which we now implicitly do via `mtl`.  Then we would have `MonadIO` independent of `base`.
This is better than having a conditional instance.

**Comment by andreasabel** [comment]
Published as 0.4.7!
